wazo_calld_client/commands/voicemails.py

```diff
@@ -57,18 +73,76 @@ def _move_message(self, url, dest_folder_id):
+        url = '{url}/{voicemail_id}/messages/{message_id}/recording'.format(
+            url=self.base_url,
+            voicemail_id=voicemail_id,
+            message_id=message_id,
+        )
         return self._get_recording(url)
 
     def get_voicemail_recording_from_user(self, message_id):
         url = self._client.url('users', 'me', 'voicemails', 'messages', message_id, 'recording')
         return self._get_recording(url)
 
+    def get_voicemail_greeting(self, voicemail_id, greeting):
+        url = '{url}/{voicemail_id}/greetings/{greeting}'.format(
+            url=self.base_url,
+            voicemail_id=voicemail_id,
+            greeting=greeting,
+        )
+        return self._get_recording(url)
+
+    def get_voicemail_greeting_from_user(self, greeting):
+        url = self._client.url('users', 'me', 'voicemails', 'greetings', greeting)
+        return self._get_recording(url)
+
+    def update_voicemail_greeting(self, voicemail_id, greeting, data):
+        url = '{url}/{voicemail_id}/greetings/{greeting}'.format(
+            url=self.base_url,
+            voicemail_id=voicemail_id,
+            greeting=greeting,
+        )
+        return self._put_recording(url, data)
+
+    def update_voicemail_greeting_from_user(self, greeting, data):
+        url = self._client.url('users', 'me', 'voicemails', 'greetings', greeting)
+        return self._put_recording(url, data)
+
+    def delete_voicemail_greeting(self, voicemail_id, greeting):
+        url = '{url}/{voicemail_id}/greetings/{greeting}'.format(
+            url=self.base_url,
+            voicemail_id=voicemail_id,
+            greeting=greeting,
+        )
+        self.session.delete(url)
+
+    def delete_voicemail_greeting_from_user(self, greeting):
+        url = self._client.url('users', 'me', 'voicemails', 'greetings', greeting)
+        self.session.delete(url)
+
+    def copy_voicemail_greeting(self, voicemail_id, greeting, dest_greeting):
+        url = '{url}/{voicemail_id}/greetings/{greeting}/copy'.format(
+            url=self.base_url,
+            voicemail_id=voicemail_id,
+            greeting=greeting,
+        )
+        self.session.post(url, json={'dest_greeting': dest_greeting})
+
+    def copy_voicemail_greeting_from_user(self, greeting, dest_greeting):
+        url = self._client.url(
+            'users', 'me', 'voicemails', 'greetings', greeting, 'copy'
+        )
+        self.session.post(url, json={'dest_greeting': dest_greeting})
+
+    def _put_recording(self, url, data):
+        r = self.session.put(
+            url, headers={'Content-type': 'audio/wav'}, data=data,
+        )
+        self.raise_from_response(r)
+
     def _get_recording(self, url):
         r = self.session.get(url, headers={'Accept': 'audio/wav'})
-        if r.status_code != 200:
-            self.raise_from_response(r)
-
+        self.raise_from_response(r)
```

Comment:
Why do you remove the `if`? Have you seen that the `raise_from_response` is overridden by `CalldCommand` and didn't check the status code before raise? (agreed it is ugly)

Reply:
/me just fall in a well hidden trap :fearful:

Reply:
Oh it's worse that I thought, voicemails does not use CalldCommand :astonished:

Reply:
I will propose a separate PR to clean this mess :smile:

Reply:
I have also rebased on top of zuul/black PR